src/main/java/nl/opengeogroep/safetymaps/server/stripes/SafetyConnectProxyActionBean.java

```diff
@@ -70,56 +83,143 @@ public Resolution proxy() throws Exception {
+            }
+
+        } catch(Exception e) {
+            return new StreamingResolution("application/json", logExceptionAndReturnJSONObject(log, "Error on " + url + "/" + path, e).toString());
+        }
+    }
+
+    private String applyAuthorizationToIncidentContent(String contentFromResponse, Connection c) throws Exception {
+        HttpServletRequest request = context.getRequest();
+
+        boolean kladbklokAlwaysAuthorized = "true".equals(Cfg.getSetting("kladblok_always_authorized", "false"));
+        boolean incidentMonitorKladblokAuthorized = kladbklokAlwaysAuthorized || request.isUserInRole(ROLE_ADMIN) || request.isUserInRole(ROLE_INCIDENTMONITOR_KLADBLOK);
+        boolean eigenVoertuignummerAuthorized = request.isUserInRole(ROLE_ADMIN) || request.isUserInRole(ROLE_EIGEN_VOERTUIGNUMMER);
+        boolean incidentMonitorAuthorized = request.isUserInRole(ROLE_ADMIN) || request.isUserInRole(ROLE_INCIDENTMONITOR);
+        
+        JSONArray content = new JSONArray(contentFromResponse);
+        JSONArray authorizedContent = new JSONArray();
+        JSONObject details = getUserDetails(request, c);
+        List<String> userVehicleList = Arrays.asList(details.optString("voertuignummer", "-").replaceAll("\\s", ",").split(","));
+
+        for(int i=0; i<content.length(); i++) {
+            JSONObject incident = (JSONObject)content.get(i);
+            JSONArray attachedVehicles;
+            
+            if (incident.has("BetrokkenEenheden")) {
+                attachedVehicles = (JSONArray)incident.get("BetrokkenEenheden");
+            } else {
+                attachedVehicles = new JSONArray();
+            }
+
+            boolean incidentForUserVehicle = false;
+            boolean authorizedForIncident = true;
+
+            for(int v=0; v<attachedVehicles.length(); v++) {
+                JSONObject vehicle = (JSONObject)attachedVehicles.get(v);
+                if (userVehicleList.contains(vehicle.get("Roepnaam"))) {
+                    incidentForUserVehicle = true;
                 }
-            };
+            }
 
-            //return new StreamingResolution(contentType.getValue(), new StringReader(content));
-        } catch(IOException e) {
-            log.error("Failed to write output:", e);
-            return null;
+            if(!incidentForUserVehicle && !eigenVoertuignummerAuthorized && incidentMonitorAuthorized && !incidentMonitorKladblokAuthorized) {
+                incident.put("Kladblokregels", new JSONArray());
+            }
+
+            if(!incidentForUserVehicle && !eigenVoertuignummerAuthorized && !incidentMonitorAuthorized) {
+                authorizedForIncident = false;
+            }
+
+            if(authorizedForIncident) {
+                authorizedContent.put(incident);
+            }
         }
+
+        return authorizedContent.toString();
+    }
+
+    private String applyAuthorizationToEenheidLocatieContent(String contentFromResponse) throws Exception {
```

Comment:
Maybe add a comment that this method filters `/eenheidlocatie` response to only return units which are assignment to an incident, because that is what it seems to be doing?